src/Security/Passwords.php

```diff
@@ -46,9 +45,7 @@ public static function hash($password, array $options = NULL)
 	 */
 	public static function verify($password, $hash)
 	{
-		return preg_match('#^\$2y\$(?P<cost>\d\d)\$(?P<salt>.{22})#', $hash, $m)
-			&& $m['cost'] >= 4 && $m['cost'] <= 31
-			&& crypt($password, $hash) === $hash;
+		return preg_match('#^\$2y\$\d\d\$.{22}#', $hash) && password_verify($password, $hash);
```

Comment:
Because `Passwords::verify` does not accept old bcrypt forms (i.e. `$2x$` or `$2a$`), but `password_verify` does.